tests/pipeline/test_adjusted_array.py

```diff
@@ -202,6 +202,85 @@ def _gen_multiplicative_adjustment_cases(dtype):
     )
 
 
+def _gen_multiplicative_adjustment_cases_with_perpsective_offset(dtype):
```

Comment:
As a reader, it's very hard to see how this is different from the other tests in the suite here.  I think it'd probably clearer and easier to just make all the tests in this suite check perspective offsets by adding that to `gen_expectations`.